fail2ban/server/filter.py

```diff
@@ -844,35 +943,50 @@ def getJournalMatch(self, match): # pragma: no cover - Base class, not used
 
 import socket
 import struct
+from .utils import Utils
 
 
 class DNSUtils:
 
 	IP_CRE = re.compile("^(?:\d{1,3}\.){3}\d{1,3}$")
 
+	# todo: make configurable the expired time and max count of cache entries:
+	CACHE_dnsToIp = Utils.Cache(maxCount=1000, maxTime=60*60)
```

Comment:
btw -- getting to "memoization" pattern, which I think would result in a better code (you could just decorate or explicitly cache any function -- thus may be other places could use it), here is my attempt at it
https://gist.github.com/yarikoptic/07bf97e320265f8fc799

BTW joblib module provides more thorough implementation of memoization, but without timeout feature and oriented for heavier arguments (numpy arrays)